services/web/server/src/simcore_service_webserver/users_bg_tasks.py

```diff
@@ -0,0 +1,89 @@
+import asyncio
+import logging
+from datetime import datetime
+from typing import AsyncIterator
+
+from aiohttp import web
+from aiopg.sa.engine import Engine
+from aiopg.sa.result import ResultProxy
+from models_library.basic_types import IdInt
+from simcore_postgres_database.models.users import UserStatus, users
+from tenacity import retry
+from tenacity.before_sleep import before_sleep_log
+from tenacity.wait import wait_exponential
+
+from ._constants import APP_DB_ENGINE_KEY
+
+logger = logging.getLogger(__name__)
+
+
+_SEC = 1
+_MINUTE = 60 * _SEC
+_HOUR = 60 * _MINUTE
+_DAY = 24 * _HOUR
+
+
+_PERIODIC_TASK_NAME = f"{__name__}.update_expired_users_periodically"
+_APP_TASK_KEY = f"{_PERIODIC_TASK_NAME}.task"
+
+
+async def update_expired_users(engine: Engine) -> list[IdInt]:
+    now = datetime.utcnow()
+    async with engine.acquire() as conn:
+        result: ResultProxy = await conn.execute(
+            users.update()
+            .values(status=UserStatus.EXPIRED)
+            .where(
+                (users.c.status == UserStatus.ACTIVE)
+                & (users.c.expires_at != None)
+                & (users.c.expires_at < now)
+            )
+            .returning(users.c.id)
+        )
+        updated_userids: list[IdInt] = await result.fetchall()
+        return updated_userids
+
+
+@retry(
+    wait=wait_exponential(min=5 * _SEC),
+    before_sleep=before_sleep_log(logger, logging.WARNING),
+)
+async def _update_expired_users_periodically(engine: Engine, wait: float = 0.5 * _DAY):
```

Comment:
Question: Why such a high interval, half a day for this check?
Just thinking: Would it not make more sense to have it on 1 hour, if there is an issue with the code it's simpler to spot with a lower interval.

Reply:
I will add it as and environment variable